github3/repos/topics.py

```diff
@@ -0,0 +1,20 @@
+# -*- coding: utf-8 -*-
+"""Topics related logic."""
+from __future__ import unicode_literals
+
+from .. import models
+
+
+class Topics(models.GitHubCore):
+    """Representation of the repository topics.
+
+    .. attribute:: names
+
+        The names of the topics.
+    """
```

Comment:
WDYT of adding an `__iter__` and a  `__len__`, so that essentially `Topics` becomes a facade of its `names` attribute? I didn't do it outright as this pattern (AFAICT) is not used elsewhere in the code base.

Reply:
I'd rather not. As you mentioned we don't have that precedent anywhere else.

Further, if GitHub adds more attributes to this endpoint (which I'm guess they will based on the structure of the JSON [1]) then iterating over the Topics object will be more confusing.

It almost feels like we shouldn't have an object for this, but again, I suspect they'll be expanding the response without telling anyone.


[1] I'm assuming that they haven't given up on consistency for the API behaviour which is what this looks like on first blush.

Reply:
> Further, if GitHub adds more attributes to this endpoint [...] then iterating over the Topics object will be more confusing.

Good point, that's an excellent argument for avoiding this magic :+1: